File: 0997-find-the-town-judge/0997-find-the-town-judge.java
class Solution {
    public int findJudge(int n, int[][] trust) {
        int arr[] = new int[n + 1];
        int in[] = new int[n + 1];

        for (int i = 0; i < trust.length; i++) {
            int id = trust[i][0];
            int num = trust[i][1];

            in[id]++;   
            arr[num]++;   
        }

        for (int i = 1; i <= n; i++) {
            if (arr[i] == n - 1 && in[i] == 0) {
                return i;
            }
        }

        return -1;
    }
}

File: 0997-find-the-town-judge/0997-find-the-town-judge.py
class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        indegree = [0] * n
        outdegree = [0] * n

        for a, b in trust:
            outdegree[a - 1] += 1
            indegree[b - 1] += 1

        for i in range(n):
            if outdegree[i] == 0 and indegree[i] == n - 1:
                return i + 1

        return -1
